### DISCUSSION TITLE
Why is this written in JavaScript?

### DISCUSSION THREAD

**Discussion by Colburse** [Dygmalab/Bazecor#409] [General]
Is there a reason JS is preferable to something like C for this kind of software?

**Comment by benjdevries** [comment]
C would be good if Bazecor were some sort of driver for the keyboard and needed to interface at a low level with the operating system, but it's really just the UI. The configuration gets written to the Neuron which controls the interface with the computer. Bazecor uses the Electron framework which lets you write web apps that run natively on pretty much any operating system, and JS is by and large the language of the web.

**Comment by javierguzman** [comment]
The firmware is done in C++ but the UI (Bazecor) is done in Electron because we need to support different platforms (Windows, Linux and MacOS). We do not have the manpower to make a native application for each platform. There are other alternatives like Tauri and we have thought about moving into it but it is still early days and I would require a lot of effort to rewrite a lot of code.
